pkg/controller/persistentvolume/persistentvolume_framework_test.go

```diff
@@ -0,0 +1,443 @@
+		claims:       make(map[string]*api.PersistentVolumeClaim),
+		ctrl:         ctrl,
+		volumeSource: volumeSource,
+		claimSource:  claimSource,
+	}
+	client.AddReactor("*", "*", reactor.React)
+	return reactor
+}
+
+func newPersistentVolumeController(kubeClient clientset.Interface) *PersistentVolumeController {
+	ctrl := &PersistentVolumeController{
+		volumes:       newPersistentVolumeOrderedIndex(),
+		claims:        cache.NewStore(cache.MetaNamespaceKeyFunc),
+		kubeClient:    kubeClient,
+		eventRecorder: &record.FakeRecorder{},
+	}
+	return ctrl
+}
+
+func newVolume(name string, capacity string, boundToClaimUID string, boundToClaimName string, phase api.PersistentVolumePhase, annotations ...string) *api.PersistentVolume {
+	volume := api.PersistentVolume{
+		ObjectMeta: api.ObjectMeta{
+			UID:             "volumeUID",
+			Name:            name,
+			ResourceVersion: "1",
+		},
+		Spec: api.PersistentVolumeSpec{
+			Capacity: api.ResourceList{
+				api.ResourceName(api.ResourceStorage): resource.MustParse(capacity),
+			},
+			PersistentVolumeSource: api.PersistentVolumeSource{
+				GCEPersistentDisk: &api.GCEPersistentDiskVolumeSource{},
+			},
+			AccessModes: []api.PersistentVolumeAccessMode{api.ReadWriteOnce, api.ReadOnlyMany},
+		},
+		Status: api.PersistentVolumeStatus{
+			Phase: phase,
+		},
+	}
+
+	if boundToClaimName != "" {
+		volume.Spec.ClaimRef = &api.ObjectReference{
+			Kind:       "PersistentVolumeClaim",
+			APIVersion: "v1",
+			UID:        types.UID(boundToClaimUID),
+			Namespace:  testNamespace,
+			Name:       boundToClaimName,
+		}
+	}
+
+	if len(annotations) > 0 {
+		volume.Annotations = make(map[string]string)
+		for _, a := range annotations {
+			volume.Annotations[a] = "yes"
+		}
+	}
+
+	return &volume
+}
+
+func newClaim(name string, claimUID string, capacity string, boundToVolume string, phase api.PersistentVolumeClaimPhase, annotations ...string) *api.PersistentVolumeClaim {
+	claim := api.PersistentVolumeClaim{
+		ObjectMeta: api.ObjectMeta{
+			Name:            name,
+			Namespace:       testNamespace,
+			UID:             types.UID(claimUID),
+			ResourceVersion: "1",
+		},
+		Spec: api.PersistentVolumeClaimSpec{
+			AccessModes: []api.PersistentVolumeAccessMode{api.ReadOnlyMany, api.ReadWriteOnce},
+			Resources: api.ResourceRequirements{
+				Requests: api.ResourceList{
+					api.ResourceName(api.ResourceStorage): resource.MustParse(capacity),
+				},
+			},
+			VolumeName: boundToVolume,
+		},
+		Status: api.PersistentVolumeClaimStatus{
+			Phase: phase,
+		},
+	}
+	// Make sure api.GetReference(claim) works
+	claim.ObjectMeta.SelfLink = testapi.Default.SelfLink("pvc", name)
+
+	if len(annotations) > 0 {
+		claim.Annotations = make(map[string]string)
+		for _, a := range annotations {
+			claim.Annotations[a] = "yes"
+		}
+	}
+	return &claim
+}
+
+func testSyncClaim(ctrl *PersistentVolumeController, reactor *volumeReactor, test controllerTest) error {
```

Comment:
They're used in deletion tests: https://github.com/kubernetes/kubernetes/pull/24331/files#diff-a9ce54ab72b21fe582583512d370585eR102